## Assignment 1 

**Student Name:** Ayah Abdalmuti

**Student ID:** 1212074

**Section:**  2

**Dataset:**  [House Prices Dataset on Kaggle](https://www.kaggle.com/c/house-prices-advanced-regression-techniques/data?select=train.csv)  🏡
      
**Number of rows:** 1460  ||  **Number of columns:** 81


### Import Libraries and Load Kaggle Dataset 

In [53]:
# Import Pandas and NumPy libraries
import pandas as pd
import numpy as np

# Load the dataset from Kaggle into a Pandas DataFrame
df = pd.read_csv('train.csv')

# Select 10 columns from the dataset for analysis
cols = ['SalePrice', 'LotArea', 'OverallQual', 'YearBuilt', 'TotalBsmtSF',
        '1stFlrSF', 'GrLivArea', 'FullBath', 'BedroomAbvGr', 'GarageCars']

# Display the first 5 rows of the selected columns
df[cols].head()

,SalePrice,LotArea,OverallQual,YearBuilt,TotalBsmtSF,1stFlrSF,GrLivArea,FullBath,BedroomAbvGr,GarageCars
0,208500,8450,7,2003,856,856,1710,2,3,2
1,181500,9600,6,1976,1262,1262,1262,2,3,2
2,223500,11250,7,2001,920,920,1786,2,3,2
3,140000,9550,7,1915,756,961,1717,1,3,3
4,250000,14260,8,2000,1145,1145,2198,2,4,3


## Summary Statistics of Selected Columns 📊 

- **Mean :**
  
  The average of the values in the SalePrice "Price of the house" column.

In [34]:
means = df['SalePrice'].mean()
print("Mean (SalePrice):", means)

Mean (SalePrice): 180921.19589041095


 This is a single representative value that summarizes all the house prices in the dataset. It shows the typical price of a house in the data. By comparing individual house prices to this mean, we can see which houses are more expensive or cheaper than the typical value, which helps in understanding the overall price distribution.

- **Standard Deviation:**
  
     Measures how spread out the values are in the LotArea "Size of the land" column.

  

In [36]:
std = df['LotArea'].std()
print("Standard Deviation (LotArea):", std)

Standard Deviation (LotArea): 9981.26493237915


This shows that the sizes of the house lots vary widely in the dataset. Some houses are on much larger lots while others are on smaller ones. A high standard deviation like this indicates that there is a lot of diversity in the land sizes among the houses.

- **Count :**
  
   The number of non-missing values in the BedroomAbvGr "Number of bedrooms above ground" column.  

In [37]:
count = df['BedroomAbvGr'].count()
print("Count (BedroomAbvGr):", count)

Count (BedroomAbvGr): 1460


This count shows that every house has data for the number of bedrooms above ground column, so there are no missing values.

- **Median :**
  
  The middle value when the data in the GrLivArea "Above-ground living area" column is sorted.

In [39]:
median = df['GrLivArea'].median()
print("Median (GrLivArea):", median)

Median (GrLivArea): 1464.0


This means that half of the houses in the dataset have an above-ground living area smaller than 1464.0 square feet, and the other half have a larger living area.  It represents a typical house size in terms of living space above ground and helps us understand the distribution of house sizes in the dataset.

- **75th Percentile :**
  
   The value below which 75% of the TotalBsmtSF "Total Basement Square Feet" data falls.


In [42]:
percentile_75 = df['TotalBsmtSF'].quantile(0.75)
print("75th Percentile (TotalBsmtSF):", percentile_75)

75th Percentile (TotalBsmtSF): 1298.25


This means that 75% of the houses in the dataset have a basement area of 1298.25 square feet or less, and the remaining 25% have a larger basement. It helps us understand the distribution of basement sizes, showing which houses have among the largest 25% of basement space in the dataset.

## Business Questions 📈

- **Question 1 :** What is the average living area (GrLivArea) of houses ?

In [49]:
living_area = df['GrLivArea'].mean()
print("Average Living Area:", living_area)

Average Living Area: 1515.463698630137


Most houses have a living area around 1515.463698630137 sq ft, helping understand how big most houses are.

- **Question 2 :** What is the typical (most common) number of full bathrooms in houses ? 

In [48]:
fullbath = df['FullBath'].median()
print("Typical number of full bathrooms:", fullbath)

Typical number of full bathrooms: 2.0


Most houses have this number of full bathrooms, helping buyers and sellers know the typical house layout and plan accordingly.

- **Question 3 :** Do houses with higher quality scores sell for higher prices ?

In [51]:
quality_price = df.groupby('OverallQual')['SalePrice'].mean()
print(quality_price)

OverallQual
1      50150.000000
2      51770.333333
3      87473.750000
4     108420.655172
5     133523.347607
6     161603.034759
7     207716.423197
8     274735.535714
9     367513.023256
10    438588.388889
Name: SalePrice, dtype: float64


Houses with higher quality scores sell for higher prices, showing a clear positive relationship : better quality → higher price. This helps buyers and sellers estimate house values based on quality. 

- **Question 4 :** What are the smallest and largest lot sizes ?

In [52]:
max_lot = df['LotArea'].max()
min_lot = df['LotArea'].min()
print("Largest Lot Area:", max_lot)
print("Smallest Lot Area:", min_lot)

Largest Lot Area: 215245
Smallest Lot Area: 1300


The largest lot is the biggest property, and the smallest lot is the tiniest. This helps buyers and sellers understand the range of land sizes available.

- **Question 5 :** How does the number of bedrooms affect house price ? 

In [54]:
price_by_bedrooms = df.groupby('BedroomAbvGr')['SalePrice'].mean()
print(price_by_bedrooms)

BedroomAbvGr
0    221493.166667
1    173162.420000
2    158197.659218
3    181056.870647
4    220421.253521
5    180819.047619
6    143779.000000
8    200000.000000
Name: SalePrice, dtype: float64


Houses with more bedrooms generally sell for higher prices. This helps buyers and sellers understand how bedroom count influences house value.